source/autocomplete/autocomplete_api.cpp

```diff
@@ -389,8 +390,28 @@ void autocomplete(navitia::PbCreator& pb_creator,
             if (search_type == 0) {
                 update_quality(found, query_word_vec.size());
             }
+
+            // Manage filter only if autocomplete result exit
+            type::Indexes indexes;
+            if (found.size() > 0) {
+                if (!ptref_filter.empty()) {
+                    try {
+                        indexes = ptref::make_query(type, ptref_filter, d);
+                    } catch (const ptref::parsing_error& parse_error) {
+                        pb_creator.fill_pb_error(pbnavitia::Error::unable_to_parse,
+                                                 "Problem while parsing the query:" + parse_error.more);
+                        return;
+                    } catch (const std::exception&) {
+                    }
+                }
+            }
```

Comment:
If we want to keep the code clean, it's nice to follow the SOLID principles. Especially the S - https://en.wikipedia.org/wiki/Single-responsibility_principle. 
A function should do one thing and a Class have one responsibility. Hence moving this into a separate function.